packages/@aws-cdk/aws-cloudfront/test/integ.cloudfront-s3.ts

```diff
@@ -12,19 +11,23 @@ const oai = new cloudfront.CfnCloudFrontOriginAccessIdentity(stack, 'OAI', {
     comment: 'Allows CloudFront to reach to the bucket!',
   }
 });
+
+const oaiImported = cloudfront.CloudFrontOriginAccessIdentity.fromName(
+  stack,
+  'OAIImported',
+  oai.ref
+);
+
 const dist = new cloudfront.CloudFrontWebDistribution(stack, 'Distribution', {
   originConfigs: [{
     behaviors: [{ isDefaultBehavior: true }],
     s3OriginSource: {
       s3BucketSource: bucket,
-      originAccessIdentityId: oai.ref,
+      originAccessIdentity: oaiImported,
     },
   }]
 });
-bucket.addToResourcePolicy(new iam.PolicyStatement({
-  actions: ['s3:Get*', 's3:List*'],
-  resources: [bucket.bucketArn, bucket.arnForObjects('*')],
-  principals: [new iam.CanonicalUserPrincipal(oai.attrS3CanonicalUserId)],
-}));
+
+bucket.grantRead(oaiImported);
```

Comment:
should this be done implicitly when adding to CloudFront?

Reply:
yes